psi4/src/psi4/libmints/matrix.cc

```diff
@@ -1387,7 +1387,23 @@ void Matrix::gemm(const char &transa, const char &transb, const int &m, const in
 void Matrix::gemm(bool transa, bool transb, double alpha, const Matrix *const a, const Matrix *const b, double beta) {
     if (nirrep_ != a->nirrep_ || nirrep_ != b->nirrep_)
         throw PSIEXCEPTION("Matrix::gemm error: Number of irreps do not equal.");
-
+    // Check matrix sizes
+    int tRows;
+    int tCols;
+    int tRowsa;
+    int tColsa;
+    int tRowsb;
+    int tColsb;
+    for (int h = 0; h < nirrep; ++h) {
+        tRows += rowspi_[h];
```

Comment:
You should never add to an uninitialized variable in C++.

Reply:
Also true.

`tRows = rowspi_.sum()` would do the same computation without a for loop (and without adding to uninitialized variables), but as discussed in my previous comment, that isn't the data validation we need.